src/wal.rs

```diff
@@ -59,7 +69,7 @@ impl Header {
     }
 
     /// Decode header from bytes
-    pub fn decode(&mut self, bytes: &mut Bytes) -> Result<usize> {
+    pub fn decode(&mut self, bytes: &[u8]) -> Result<usize> {
```

Comment:
This should break implementation of iterator, which depends on `decode` mutating `Bytes`.

Reply:
We could always convert `&Bytes` to `&[u8]` without overhead. And when we're decoding header from `EntryRef`, we had to use `&[u8]`. So I think this change is reasonable.

Reply:
I'm not objecting the changes. I'm suggesting the change introduces a bug at https://github.com/tikv/agatedb/blob/afb59d2b2cf294408ceb0b186d08b6a19dfb662c/src/table/iterator.rs#L123-L139 that needs to be fixed.

Reply:
That's not the same header. I think this naming might be confusing. This is the header for WAL, and the one in iterator is the SST header. I think we could later unify these interfaces :)

Reply:
What about using `TableHeader` for SST, and `WalHeader` for write-ahead log / vlog? Meanwhile, we could change all `decode` to use `&[u8]` as parameter to reduce the need of using `slice` on `Bytes` (which introduces overhead of increasing the reference counter)

Reply:
LGTM, but better another PR.

Reply:
Alright. I've reverted this change.